packages/tabs/src/InternalTab.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import Portal from '@leafygreen-ui/portal';
+import { IdAllocator } from '@leafygreen-ui/lib';
+import TabTitle from './TabTitle';
+import { TabsProps } from './Tabs';
+
+const tabIdAllocator = IdAllocator.create('tab');
+const tabpanelIdAllocator = IdAllocator.create('tab-panel');
+
+type InternalTabProps = Pick<TabsProps, 'as' | 'darkMode' | 'className'> & {
+  child: React.ReactElement;
+  onKeyDown: (e: KeyboardEvent) => void;
+  onClick?: (e: React.MouseEvent) => void;
+  isAnyTabFocused?: boolean;
+  selected: boolean;
+  tabRef: HTMLDivElement | null;
+  panelRef: HTMLDivElement | null;
+};
+
+const InternalTab = React.memo(
+  ({
+    child,
+    selected,
+    as,
+    tabRef,
+    panelRef,
+    ...tabProps
+  }: InternalTabProps) => {
+    const { id: idProp, name } = child.props;
+
+    const panelId = React.useMemo(() => tabpanelIdAllocator.generate(), []);
```

Comment:
@lazytype upon further work with the component, you were totally right that changing this on every render is a bug.